### PR TITLE
 Fix freezing of map fields.

### PR DESCRIPTION
**Pull request by szakarias**

### CODE REVIEW

File: protoc_plugin/test/map_field_test.dart [hunk 1]
```diff
@@ -267,4 +267,39 @@ void main() {
     testMap.mergeFromBuffer(testMap2.writeToBuffer());
     expect(testMap.int32ToMessageField[2].secondValue, 42);
   });
+
+  test('Freeze message with map field', () {
+    TestMap testMap = TestMap();
+    _setValues(testMap);
+    testMap.freeze();
+
+    try {
```
Comment (sigurdm) [inline protoc_plugin/test/map_field_test.dart]: You can write these tests of throwing a message better with:
```
expect(() => doStuffTheShouldThrow(), throwsA(const TypeMatcher<UnsupportedError>()))
```

File: protobuf/lib/src/protobuf/pb_map.dart [hunk 2]
```diff
@@ -31,26 +32,37 @@ class PbMap<K, V> extends MapBase<K, V> {
...
+  }
 
   @override
   Iterable<K> get keys => _wrappedMap.keys;
 
   @override
-  V remove(Object key) => _wrappedMap.remove(key);
+  V remove(Object key) {
+    if (_isReadonly)
+      throw new UnsupportedError('Attempted to change a read-only map field');
```
Comment (sigurdm) [inline protobuf/lib/src/protobuf/pb_map.dart]: Ideally this would hook into the `frozenMessageModificationHandler`

File: protobuf/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 0.10.8
+
+* Add freeze support for map fields.
```
Comment (sigurdm) [inline protobuf/CHANGELOG.md]: Perhaps this should be phrased as:
```
* Fix freezing of map fields.
```

File: protobuf/lib/src/protobuf/pb_map.dart [hunk 4]
```diff
@@ -31,26 +32,37 @@ class PbMap<K, V> extends MapBase<K, V> {
...
+    _wrappedMap.remove(key);
```
Comment (sigurdm) [inline protobuf/lib/src/protobuf/pb_map.dart]: You need to return the value here

File: protobuf/lib/src/protobuf/pb_map.dart (hunk 2 above)
Comment (szakarias) [reply protobuf/lib/src/protobuf/pb_map.dart]: Leaving this out, per discussion.

File: protoc_plugin/test/map_field_test.dart (hunk 1 above)
Comment (szakarias) [reply protoc_plugin/test/map_field_test.dart]: Done

File: protobuf/CHANGELOG.md (hunk 3 above)
Comment (szakarias) [reply protobuf/CHANGELOG.md]: Done

File: protobuf/lib/src/protobuf/pb_map.dart (hunk 4 above)
Comment (szakarias) [reply protobuf/lib/src/protobuf/pb_map.dart]: Yes, thanks. Done.

**Pull request merged**